apps/docs/scripts/generate-embeddings.ts

```diff
@@ -0,0 +1,335 @@
+import { createClient } from '@supabase/supabase-js'
+import { createHash } from 'crypto'
+import dotenv from 'dotenv'
+import { ObjectExpression } from 'estree'
+import { readdir, readFile, stat } from 'fs/promises'
+import { Content, Root } from 'mdast'
+import { fromMarkdown } from 'mdast-util-from-markdown'
+import { mdxFromMarkdown, MdxjsEsm } from 'mdast-util-mdx'
+import { toMarkdown } from 'mdast-util-to-markdown'
+import { mdxjs } from 'micromark-extension-mdxjs'
+import 'openai'
+import { Configuration, OpenAIApi } from 'openai'
+import { join } from 'path'
+import { u } from 'unist-builder'
+import { filter } from 'unist-util-filter'
+import { inspect } from 'util'
+
+dotenv.config()
+
+const ignoredFiles = ['pages/404.mdx']
+
+/**
+ * Extracts ES literals from an `estree` `ObjectExpression`
+ * into a plain JavaScript object.
+ */
+function getObjectFromExpression(node: ObjectExpression) {
+  return node.properties.reduce<
+    Record<string, string | number | bigint | true | RegExp | undefined>
+  >((object, property) => {
+    if (property.type !== 'Property') {
+      return object
+    }
+
+    const key = (property.key.type === 'Identifier' && property.key.name) || undefined
+    const value = (property.value.type === 'Literal' && property.value.value) || undefined
+
+    if (!key) {
+      return object
+    }
+
+    return {
+      ...object,
+      [key]: value,
+    }
+  }, {})
+}
+
+/**
+ * Extracts the `meta` ESM export from the MDX file.
+ *
+ * This info is akin to frontmatter.
+ */
+function extractMetaExport(mdxTree: Root) {
+  const metaExportNode = mdxTree.children.find((node): node is MdxjsEsm => {
+    return (
+      node.type === 'mdxjsEsm' &&
+      node.data?.estree?.body[0]?.type === 'ExportNamedDeclaration' &&
+      node.data.estree.body[0].declaration?.type === 'VariableDeclaration' &&
+      node.data.estree.body[0].declaration.declarations[0]?.id.type === 'Identifier' &&
+      node.data.estree.body[0].declaration.declarations[0].id.name === 'meta'
+    )
+  })
+
+  if (!metaExportNode) {
+    return undefined
+  }
+
+  const objectExpression =
+    (metaExportNode.data?.estree?.body[0]?.type === 'ExportNamedDeclaration' &&
+      metaExportNode.data.estree.body[0].declaration?.type === 'VariableDeclaration' &&
+      metaExportNode.data.estree.body[0].declaration.declarations[0]?.id.type === 'Identifier' &&
+      metaExportNode.data.estree.body[0].declaration.declarations[0].id.name === 'meta' &&
+      metaExportNode.data.estree.body[0].declaration.declarations[0].init?.type ===
+        'ObjectExpression' &&
+      metaExportNode.data.estree.body[0].declaration.declarations[0].init) ||
+    undefined
+
+  if (!objectExpression) {
+    return undefined
+  }
+
+  return getObjectFromExpression(objectExpression)
+}
```

Comment:
I noticed we do this already in `build-search.ts` (using string parsing ops like slicing, `indexOf`, regex, etc), but now that we have access to the `mdast`, it seemed right to grab this directly from the source. Hope that's alright.